### DISCUSSION TITLE
Data Replication

### DISCUSSION THREAD

**Discussion by lisapatel12** [gramps-project/gramps-web#132] [Announcements]
Is data replication compatible with gramp.js? If so, may you please further elaborate on how it can be done or if you have an open source code ready for it?

**Comment by DavidMStraub** [comment]
Can you please elaborate what exactly you mean? There is the XML export feature and the desktop sync feature for instance.

**Comment by lisapatel12** [comment]
For example, if I upload data for one person on Device A, when I open up device B, the data is already loaded and replicated onto any devices that I use if that makes sense. Another example would be  I need access to my todo list on my phone while I walk to the office. While at work, on my laptop, I should be able to view the tasks I’ve entered on my phone and vice-versa. 

What would be the most efficient way to do that in gramps.js? Would it require use of an Oracle database or between two relational data stores like SQL Server and PostgreSQL?

**Comment by DavidMStraub** [comment]
Gramps.js is the frontend for [Gramps Web](https://gramps-project.github.io/web/) which is a complete web app with a REST API backend called [Gramps Web API](https://github.com/gramps-project/gramps-webapi/). So accessing it from different devices is definitely one of the main use cases.

**Comment by lisapatel12** [comment]
So if I use Gramps Web API, I would be able to sync and replicate data providing access to different devices? Is there already a source code written for this type of use? Also since Gramps Web API would ideally be used, would Oracle or others be used?

Update:
**Gramps Web API Sync Addon**
This would be the most ideal thing to use right??^^

**Comment by DavidMStraub** [comment]
It depends on what you want to achieve.

You can use it only in the browser - in that case, no additional tool is needed.

You can get full backups anytime by downloading an XML export.

If you want to use Gramps desktop on your PC and Gramps Web in your browser, then yes, you have to use the sync addon.
